### PR TITLE
UPDATE milestone complete

### PR DESCRIPTION
**Pull request by elliewalkersm**
## Description
Completed the Update milestone. When authenticated and logged in, user can update items for staff, reservations, ingredients and menu. The items update patch the data in Firebase and the DOM is dynamically rendered after each item is updated.

## Link to deployed app
https://development--michelin-star-restaurant.netlify.app/#

## Related Issue
#25 
#27 
#28 
#30 

## Motivation and Context
The user can update the data of an existing item.

## How Can This Be Tested?
visit the deployed link, login, and click on the edit button on any item. A form will appear to update the values for that item. Click update and the dom will show the updated item.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: src/javascripts/components/menu/menu.js [hunk 1]
```diff
@@ -10,10 +10,10 @@ const showMenuItems = (array) => {
     <div class="media-body">
       <h5 class="mt-0">${item.title}</h5>
       <p>${item.description}</p>
-      <p>${item.price}</p>
       <div class="pb-2">
-      <button type="button" class="btn btn-info" data-toggle="modal" data-target="#formModal" id="view-menu-ingredients--${item.firebaseKey}">View Ingredients</button>
+      <button type="button" class="btn btn-link" data-toggle="modal" data-target="#formModal" id="view-menu-ingredients--${item.firebaseKey}">Show Full Ingredient List</button>
```
Comment (drteresavasquez) [inline src/javascripts/components/menu/menu.js:14]: Try and be consistent with the id separators. Choose either `--` or `^^`

**Review by drteresavasquez — COMMENTED** [review]
Create a bug ticket for when I log in and then log out and create a staff member, they show up twice. I am not sure if this is happening in other views, but it is definitely happening there.

https://www.loom.com/share/0ea854d566e44a2987a4327339c84c79

**Review by TrinityChristiana — COMMENTED** [review]
Menu Items
- Ingredients is not populating with values of the ingredients already to the Menu Item
    - If I want to edit a menu item that already has `"bread"` as an ingredient, then `"bread"` should already be selected
- Available checkbox is not populating with the current value
    - If I want to edit a menu item that already is `available`  then `available` should already be selected

Ingredients
- Form is not populating with the current value of the ingredient. 
    - If I want to edit `"Bread"`, the input should have the text `"Bread"` to start out.

Staff
- All good!

Reservations
- All good

Bug ticket to create for later (_You don't have to fix this to get approved, just create a bug ticket for this to fix later_)
- Found a bug where if you start creating an ingredient and click outside of the modal, then the button to create an ingredient doesn't pop back up.

https://user-images.githubusercontent.com/31781724/111880154-a9ee1c80-8977-11eb-8a99-dbdfbb6c6736.mov

**Comment by TrinityChristiana** [comment]
1. The checkboxes with the ingredients show up with the correct values now!!! We just need to get the Available checkbox to do the same thing. 
1. Now, if I edit a menu item and select available, then it adds the value of the checkbox to the ingredient array on the menu. An example is below.
    - Screen Shot 2021-03-22 at 2 24 04 PM
1. We also still need for the Ingredient edit form to be populated with the current value of the ingredient.

**Review by TrinityChristiana — APPROVED** [review]
Nice!

**Review by caseywalker — APPROVED** [review]

**Pull request merged**